### ISSUE TITLE
ACT policy lower than expected performance in aloha-gym

### ISSUE THREAD

**Issue by aposadasn** [huggingface/lerobot#2605]
### System Info

```Shell
- lerobot version: 0.4.3
- Platform: Linux-6.14.0-36-generic-x86_64-with-glibc2.39
- Python version: 3.10.19
- Huggingface Hub version: 0.35.3
- Datasets version: 4.1.1
- Numpy version: 2.2.6
- PyTorch version: 2.7.1+cu126
- Is PyTorch built with CUDA support?: True
- Cuda version: 12.6
- GPU model: NVIDIA GeForce RTX 4080 Laptop GPU
- Using GPU in script?: yes
```

### Information

- [x] One of the scripts in the examples/ folder of LeRobot
- [ ] My own task or dataset (give details below)

### Reproduction

I am trying to use the ACT policy in the aloha-gym env. I ran the following script:
lerobot-train
--policy.type=act
--dataset.repo_id=lerobot/aloha_sim_transfer_cube_human
--env.type=aloha
--env.task=AlohaTransferCube-v0
--wandb.enable=true
--policy.repo_id=aposadasn/act_aloha_sim_transfer_cube_human_model

Eval with:
lerobot-eval \
    --policy.path=aposadasn/act_aloha_sim_transfer_cube_human_model \
    --env.type=aloha \
    --env.task=AlohaTransferCube-v0 \
    --eval.n_episodes=500 \
    --policy.device=cuda

### Expected behavior

On the HF lerobot site: https://huggingface.co/lerobot/act_aloha_sim_transfer_cube_human the model which was based on an older version of lerobot benchmarks performance at 83% success rate over 500 tasks. The training curves also confirm this: https://wandb.ai/aliberts/lerobot/runs/720l37xb.

However, in the latest version of lerobot, I do not reach this level of performance. I have tried on 2 different machines and come to the same result

<img width="2402" height="662" alt="Image" src="https://github.com/user-attachments/assets/f5dbe4c9-aa64-4d40-a7c5-c6e41c1b2f55" />

When I run evaluation over 500 episodes using this policy I get success rates around 50-60%. I am not sure where the problem lies but I suspect it's the dataset.

**Comment by aposadasn** [comment]
Btw I think this issue also applies to the Aloha Insertion task.

**Comment by atyshka** [comment]
The aloha sim is pretty bad... the oracle script for ACT insertion has about a 60% failure rate, and the default ACT code doesn't filter out bad demonstrations. I'm not sure whether the lerobot implementation does.

**Comment by aposadasn** [comment]
The result that lerobot posted and the result from the original ACT paper should be reproducible though given the same dataset and model. The issue is I cannot get a similar result for either task using the same dataset and model.

**Comment by atyshka** [comment]
It's possible HF used the correct humber of decoder layers in their own benchmark of ACT. If you check the LeRobot config, decoder layers are set to 1 because there was a bug in the original ACT implementation where the supposed 7 decoder layers were actually only one layer. Try setting it higher, like 6 or 7, and I bet you'll achieve higher success rate.

**Comment by aposadasn** [comment]
This is the policy configuration from the [wandb run](https://wandb.ai/aliberts/lerobot/runs/720l37xb/files/config.yaml) that the lerobot team posted, the number of decoder layers is set to 1 just as in the ACT repo. Btw I am not using the ACT repo I only use lerobot.

policy:
    value:
        chunk_size: 100
        dim_feedforward: 3200
        dim_model: 512
        dropout: 0.1
        feedforward_activation: relu
        kl_weight: 10
        latent_dim: 32
        n_action_steps: 100
        n_decoder_layers: 1
        n_encoder_layers: 4
        n_heads: 8
        n_obs_steps: 1
        n_vae_encoder_layers: 4
        normalization_mapping:
            ACTION: MEAN_STD
            STATE: MEAN_STD
            VISUAL: MEAN_STD
        optimizer_lr: 1e-05
        optimizer_lr_backbone: 1e-05
        optimizer_weight_decay: 0.0001
        pre_norm: false
        pretrained_backbone_weights: ResNet18_Weights.IMAGENET1K_V1
        replace_final_stride_with_dilation: false
        temporal_ensemble_coeff: null
        use_vae: true
        vision_backbone: resnet18

With this config, the lerobot team got the following result at step 80K:
INFO 2025-01-25 15:50:37 ts/train.py:172 step:80K smpl:640K ep:2K epch:32.00 ∑rwrd:214.900 success:74.0% eval_s:237.302 

On the version posted above of lerobot, I am only able to get:
INFO 2025-12-06 15:09:32 ot_train.py:408 Suite overall aggregated: {'avg_sum_reward': 148.9, 'avg_max_reward': 2.42, 'pc_success': 38.0, 'n_episodes': 50, 'eval_s': 147.07790327072144, 'eval_ep_s': 2.9415580654144287}

My issue is I use the same exact config, same exact dataset with much lower success rate in the aloha sim env.

**Comment by reeceomahoney** [comment]
I've found a similar issue with the insertion task. I only get 8% success vs the 20% they report. And the original version on huggingface can't be evaluated on main as its missing processor configs

**Comment by aposadasn** [comment]
@reeceomahoney if you're interested in tasks like the ones in aloha that are easy to run with the lerobot api and have human-generated datasets with teleoperation, check out this benchmark: https://github.com/ARCLab-MIT-X/beavr-bench

**Comment by imstevenpmwork** [comment]
Hello guys! Curious to know if this is still happening and happy to help solving it

**Comment by shuowoshishui** [comment]
I encountered the same issue in my own Isaac task. With the same dataset and environment, the ACT policy in Aloha achieves a 90% success rate, but the current LeRobot version only reaches 34%.

**Comment by shuowoshishui** [comment]
@imstevenpmwork  I encountered the same issue in my own Isaac task. With the same dataset and environment, the ACT policy in Aloha achieves a 90% success rate, but the current LeRobot version only reaches 34%.
